.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+closes #???
+
+The following tasks have been completed:
+
+ * [ ] Confirmed there are no ReSpec errors/warnings.
+ * [ ] Modified Web platform tests (link)
```

Comment:
I think it's too late by that point... I would encourage you to not merge things without tests.

Reply:
This was a tradeoff we landed on to after consulting @foolip on the matter. That is to say, we’re happy to have tests for every PR but won’t block PRs for specs earlier in their lifecycle.

Reply:
It shouldn't... but you need to ship your thinking a little: the "early in their lifecycle" is basically done in the pull request. Then you rely on pr-preview + the pull request itself to develop the feature, together with tests and actual implementation.... the end result is that you never end up wit anything in the spec that's not fully tested, implemented, and with full implementation commitments. (see WHATWG work mode... and see also Payment Request API). Yes, this means it can take months to land a PR... and some PRs never land.... but you never end up with random experimental things in the spec, which is actually really nice.

Reply:
The "in CR or above" bit I believe was added to make it clear that for specs that are very early on and don't have any implementation yet it's not always a good idea to write tests.

The transition from one working mode for the other is tricky, but I assume this template is only for specs where it already makes sense to require tests, etc.?

Reply:
> The "in CR or above" bit I believe was added to make it clear that for specs that are very early on and don't have any implementation yet it's not always a good idea to write tests.

Yeah, the assumption is that by the time this hits a working group the idea should have been incubated. So reconstructing a spec from incubation presumes a whole lot of stability. If not, then the incubation should happen as part of the pull request.  

> The transition from one working mode for the other is tricky, but I assume this template is only for specs where it already makes sense to require tests, etc.?

Yes. Though I don't see how any spec cannot require tests? IMO, we shouldn't put half baked things into specs if we can avoid it - specially for specs that are highly testable, like this one.

Reply:
> I don't see how any spec cannot require tests? 

I think it's reasonable in the very first phase of a spec's life to first write a bunch of spec text, then separate implement and write a bunch of tests for it. Possible even in a few iterations over a couple of months. Problem is how to transition for that prototyping working mode to one where one can be sure that the spec has implementation commitments and tests. Best suggestion I have is to vet the spec and test suite once and then add a README and PR template like this.

Reply:
> Problem is how to transition for that prototyping working mode to one where one can be sure that the spec has implementation commitments and tests.

This is a problem of socializing the idea... which is not really a problem: the editors need to pitch the idea to the working group and those implementing and get everyone on board. So it's just consensus building.